File: DoiTuyen/Practice/Ksat tháng 4/Test 20.04.2024/BSBB/BSBB.cpp
#include <bits/stdc++.h>

using namespace std;
int n,ans = 0;
int cal(int x)
{
    if (x == 1) return 1;
    int i = 2;
    int count_div = 1; ///luu so uoc cua x
    while (i*i <= x)
    {
        int d = 0; /// luu so mu cua so nguyen to i
        while (x % i == 0)
        {
            d++;
            x = x/i;
        }
        i++;
        count_div *= (d+1);
    }
    if (x > 1) count_div *= 2;
    return count_div;
}
int main()
{
    freopen("bsbb.inp","r",stdin);
    freopen("bsbb.out","w",stdout);
    cin >> n;
    for (int i=1; i<n; i++)
        ans += cal(i);
    cout << ans;
    return 0;
}

File: DoiTuyen/Practice/Ksat tháng 4/Test 20.04.2024/BSBB/BSBB.py
fi = open("bsbb.inp","r")
n = int(fi.readline())
fi.close()

def cal(x):
    if (x == 1): return 1
    i = 2
    count_div = 1
    while (i*i <= x):
        d = 0
        while (x % i == 0):
            d += 1
            x = x // i
        count_div *= d + 1
        i += 1
    if (x > 1): count_div *= 2
    return count_div

ans = 0
for i in range(1,n):
    ans += cal(i)
fo = open("bsbb.out","w")
fo.write(str(ans))
fo.close()